test/framework/src/main/java/org/elasticsearch/index/mapper/AbstractScriptFieldTypeTestCase.java

```diff
@@ -198,6 +199,11 @@ public final void testOnScriptErrorFail() throws IOException {
         }
     }
 
+    public void testScriptFieldAlwaysHaveValue() {
+        MappedFieldType fieldType = build("loop", Map.of(), OnScriptError.FAIL);
+        assertTrue(fieldType.fieldHasValue(FieldInfos.EMPTY));
+    }
```

Comment:
You could add a test method to FieldTypeTestCase, that tests the default behaviour of all field types, and requires to be overridden for the three that have a different behaviour. Perhaps this could replace the MappedFieldTypeTests you added too?

Reply:
Done!
For `AbstractScriptFieldType` and `RankFeatureMetaFieldMapperTests` I could not because it's not extending `FieldTypeTestCase` but `MapperServiceTestCase`.
For All constant I created a new class `ConstantFieldTypeTestCase` to avoid code dup.

Reply:
Found a solution.